images/sds-drbd-controller/pkg/controller/drbd_storage_class_watcher.go

```diff
@@ -0,0 +1,349 @@
+		switch dsc.Spec.Replication {
+		case ReplicationNone:
+		case ReplicationAvailability, ReplicationConsistencyAndAvailability:
+			nodes := spNodes[dsc.Spec.StoragePool]
+			zoneNodesCount := make(map[string]int, len(nodes))
+			for _, node := range nodes {
+				if zone, exist := node.Labels[ZoneLabel]; exist {
+					zoneNodesCount[zone]++
+				}
+			}
+			log.Debug(fmt.Sprintf("[ReconcileDRBDStorageClassReplication] DRBDStorageClass %s topology type %s", dsc.Name, dsc.Spec.Topology))
+			switch dsc.Spec.Topology {
+			// As we need to place 3 storage replicas in a some random zone, we check if at least one zone has enough nodes for quorum.
+			case TopologyZonal:
+				var enoughNodes bool
+				for _, nodesCount := range zoneNodesCount {
+					if nodesCount > 2 {
+						enoughNodes = true
+					}
+				}
+
+				if !enoughNodes {
+					err := errors.New("not enough nodes in a single zone for a quorum")
+					log.Error(err, fmt.Sprintf("[ReconcileDRBDStorageClassReplication] replicas validation failed for DRBDStorageClass %s", dsc.Name))
+
+					setNonOperationalLabelOnStorageClass(ctx, cl, log, dsc, NonOperationalByReplicasLabel)
+				} else {
+					removeNonOperationalLabelOnStorageClass(ctx, cl, log, dsc, NonOperationalByReplicasLabel)
+				}
+				// As we need to place every storage replica in a different zone, we check if at least one node is available in every selected zone.
+			case TopologyTransZonal:
+				enoughNodes := true
+				for _, zone := range dsc.Spec.Zones {
+					nodesCount := zoneNodesCount[zone]
+					if nodesCount < 1 {
+						enoughNodes = false
+					}
+				}
+
+				if !enoughNodes {
+					err := errors.New("not enough nodes are available in the zones for a quorum")
+					log.Error(err, fmt.Sprintf("[ReconcileDRBDStorageClassReplication] replicas validation failed for DRBDStorageClass %s", dsc.Name))
+
+					setNonOperationalLabelOnStorageClass(ctx, cl, log, dsc, NonOperationalByReplicasLabel)
+				} else {
+					removeNonOperationalLabelOnStorageClass(ctx, cl, log, dsc, NonOperationalByReplicasLabel)
+				}
+				// As we do not care about zones, we just check if selected storage pool has enough nodes for quorum.
+			case TopologyIgnored:
+				if len(spNodes[dsc.Spec.StoragePool]) < 3 {
+					err := errors.New("not enough nodes are available in the zones for a quorum")
+					log.Error(err, fmt.Sprintf("[ReconcileDRBDStorageClassReplication] replicas validation failed for DRBDStorageClass %s", dsc.Name))
+
+					setNonOperationalLabelOnStorageClass(ctx, cl, log, dsc, NonOperationalByReplicasLabel)
+				} else {
+					removeNonOperationalLabelOnStorageClass(ctx, cl, log, dsc, NonOperationalByReplicasLabel)
+				}
+			}
+		default:
+			err := errors.New("unsupported replication type")
+			log.Error(err, fmt.Sprintf("[ReconcileDRBDStorageClassReplication] replication type validation failed for DRBDStorageClass %s", dsc.Name))
+
+			setNonOperationalLabelOnStorageClass(ctx, cl, log, dsc, "storage.deckhouse.io/non-operational")
```

Comment:
Why not ?
```suggestion
			setNonOperationalLabelOnStorageClass(ctx, cl, log, dsc, NonOperationalByReplicasLabel)
```

Reply:
The label says there are not enough nodes in the selected zones, but in the default case we face an unsupported replication type, so the message about "not enough zones" is not relevant in this case. 
As long as the case with unsupported replication type is about to never happens, I decided not to create a special label but use some general one instead.